### PR TITLE
add luvit thread support.

### PR DESCRIPTION
**Pull request by zhaozg**
based on luv https://github.com/luvit/luv/commit/e536f14103f4a6abdda9a8b070bbc95f1511278d,
luvi https://github.com/luvit/luvi/commit/20142c2ce8be56b6bbb99d466ef87f27cd2356de,
and please see examples/multi-thread-tick.lua to know how to use

### CODE REVIEW

File: deps/thread.lua [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+--]]
+
+--- luvit thread management
+local uv = require('uv')
+local base = require('luvi').bundle.base
+
+exports.name = "luvit/thread"
+exports.version = "0.0.1"
+exports.dependencies = {
+}
```
Comment (creationix) [inline deps/thread.lua]: No need for an empty table if there are no dependencies.  The field is optional.

**Comment by rphillips** [comment]
can we add a test for this? or do we need a new luvi first?

**Comment by creationix** [comment]
Why is the luvi logic inlined in this (bundle and path)?  Was this because you're waiting on a custom luvi build with the thread patch there too?

**Comment by zhaozg** [comment]
Test need more time to code.please wait I to finish test code.

**Comment by zhaozg** [comment]
@creationix I think thread should can use luvit deps lib. So need to use luvit require methods.

**Comment by zhaozg** [comment]
Any simpler solution ?

**Comment by zhaozg** [comment]
This need update luvi to work

**Comment by creationix** [comment]
I thought the purpose of the luvi patch was to give threads access to the luvi environment without having to copy the luvi libraries?

**Comment by creationix** [comment]
I found a way to clean this up.  It wasn't easy.

The luvi changes are in https://github.com/luvit/luvi/pull/118

And this PR needs the following changes https://github.com/luvit/luvit/commit/88280f938d64a4daf09b96bf4e27397dfb730c5d

**Comment by creationix** [comment]
Moving to #799

**Pull request closed without merging**